core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/uri/ComponentURI.java

```diff
@@ -113,6 +113,10 @@ private ComponentURI(CodeSystemURI codeSystem, short terminologyComponentId, Str
 		this.identifier = identifier;
 	}
 	
+	public static ComponentURI of(CodeSystemURI codeSystemURI, short terminologyComponentId, String identifier) {
+		return new ComponentURI(codeSystemURI, terminologyComponentId, Strings.nullToEmpty(identifier));
```

Comment:
Null to empty call is really bad here, both `null` and `empty` are invalid values for the identifier part. Don't convert it to empty here, validate instead on the other factory method.